### ISSUE TITLE
L-moments and L-moment ratios in .stdmetrics

### ISSUE THREAD

**Issue by redfoxgis** [Jean-Romain/lidR#317]
Useful grid metrics for modeling forest attributes are L-moments and L-moment ratios. Currently, I am only aware of these metrics available in FUSION's grid metrics and cloud metrics functions. Are there any plans to include these in lidR's .stdmetrics?

Here is an paper showing the importance of L-moments in predicting forest fuels attributes (Figure 2):
https://www.mdpi.com/2571-6255/2/3/38

Here is the FUSION documentation which mentions use of L-moments:
http://forsys.cfr.washington.edu/Software/FUSION/FUSION_manual.pdf

**Comment by Jean-Romain** [comment]
I did not even know about the term L-moment. Now I know. I guess one day or another I will have to create `.stdfusionmetrics`. It seems it is a recurrent request. But L-moment seems to have an untrivial definition:

https://github.com/cran/EnvStats/blob/master/R/lMoment.R#L1
https://github.com/cran/Lmoments/blob/master/R/Lmoments.R#L1

I guess I should take the [sample L-moment](https://en.wikipedia.org/wiki/L-moment#Sample_L-moments).

**Comment by tgoodbody** [comment]
JR - I am looking to calculate these metrics as well. Judging by the rejected request tag I assume there is no implementation of this in lidR yet? May be worth another look as I am hearing about more users wanting to calculate L- metrics.

**Comment by Jean-Romain** [comment]
The request was more than a year old and I did not find a single a day where I was motivated in working on that topic and it is not going to change. The request is good, simply I have many more thrilling feature to do so I gave up on this one. I don't know the which definition of L-moment I should use and I don't know which definition FUSION is using.

With external help the situation may change and I could do it.

**Comment by tgoodbody** [comment]
Fair enough, sounds good.

**Comment by tgoodbody** [comment]
Wow - thanks JR this is great... Appreciate you taking the time

**Comment by rubenvalpue** [comment]
This is brilliant JR. Sorry, I just didn't know it was possible to do that, as I haven't worked that much with lidR. Definitely a working solution. I take back my 'critical miss' wording. I hope you didn't find it too harsh, but apologize if you did as it was not my intention. 
I agree you should keep dependencies to the minimum. They gave much headache after the last R version release, and they are prone to always give more. The more stand-alone the better.
Great work JR!

**Comment by Jean-Romain** [comment]
No problem. I understand that there is a lot of documentation and people do not read every single document (but it is definitively documented in `grid_metrics()` and all other `metrics` related manual pages). I think it deserves an addition in the book somewhere in the "metrics¨ chapters. L-moment + fractal dimension as [suggested earlier](https://github.com/Jean-Romain/lidR/discussions/428) (I gave the same answer than here). @tgoodbody  any thought on the best place to put that?

**Comment by rubenvalpue** [comment]
Thanks a lot Jean-Romain

**Comment by tgoodbody** [comment]
JR - I will take a look now. I think it makes most sense in the `grid_metrics()` section. I need to review section 16 anyways :)

**Comment by Jean-Romain** [comment]
Yes definitively. Those metrics have no meaning at other regularization level

**Comment by rubenvalpue** [comment]
It may make sense to some in cloud_metrics too. Not for my stratification applications. But some people use them in the modelling too, like in that Mpdi paper on top of the thread. They are very stable moments, so they have been working particularly well at low densities. I think that's why.
Let me know if any help needed @tgoodbody

**Comment by Jean-Romain** [comment]
Yes indeed. To me `cloud_metrics()` is only a convenient function used to develop models on plots but then metrics are truly computed with `grid_metrics()`. In that sense `cloud_metrics` is not really a tool in lidR. But it only my interpretation base on my usage.

We could put that in chap 8 which is short . Section [8.2 user defined metrics](https://jean-romain.github.io/lidRbook/metrics.html#metrics-user-defined) or in a new section 8.4 example of complex metrics using third party packages

**Comment by tgoodbody** [comment]
Sounds good JR. I will write something up and push it later today for you to review.

**Comment by tgoodbody** [comment]
@Jean-Romain I've created section 8.4 and added brief text about 3rd party package integration. I used the 2 examples discussed here (lmom and fractional dimensions) and gave some rationale for wanting to reduce the number of dependencies lidR has. Take a look at the edits I've pushed and let me know if you want anything changed.

Thanks @rubenvalpue for your comments and @redfoxgis for getting this conversation started! 👍
